src/DropdownMenu/DropdownMenu.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {useCallback, useRef, useState} from 'react'
+import {List, GroupedListProps, ListPropsBase, ItemInput} from '../ActionList/List'
+import Overlay from '../Overlay'
+import {DropdownButton, DropdownButtonProps} from './DropdownButton'
+import {Item} from '../ActionList/Item'
+import {useFocusTrap} from '../hooks/useFocusTrap'
+import {useFocusZone} from '../hooks/useFocusZone'
+import {useAnchoredPosition} from '../hooks/useAnchoredPosition'
+import {useRenderForcingRef} from '../hooks/useRenderForcingRef'
+import randomId from '../utils/randomId'
+
+export interface DropdownMenuProps extends Partial<Omit<GroupedListProps, keyof ListPropsBase>>, ListPropsBase {
+  /**
+   * A custom function component used to render the anchor element.
+   * Will receive the selected text as `children` prop when an item is activated.
+   * Uses a `DropdownButton` by default.
+   */
+  renderAnchor?: <T extends React.HTMLAttributes<HTMLElement>>(props: T) => JSX.Element
+
+  /**
+   * A placeholder value to display on the trigger button when no selection has been made.
+   */
+  placeholder?: string
+
+  /**
+   * An `ItemProps` item from the list of `items` which is currently selected.  This item will receive a checkmark next to it in the menu.
+   */
+  selectedItem?: ItemInput
+
+  /**
+   * A callback which receives the selected item or `undefined` when an item is activated in the menu.  If the activated item is the same as the current
+   * `selectedItem`, `undefined` will be passed.
+   */
+  onChange?: (item?: ItemInput) => unknown
+}
+
+/**
+ * A `DropdownMenu` provides an anchor (button by default) that will open a floating menu of selectable items.  The menu can be
+ * opened and navigated using keyboard or mouse.  When an item is selected, the menu will close and the `onChange` callback will be called.
+ * If the default anchor button is used, the anchor contents will be updated with the selection.
+ */
+export function DropdownMenu({
```

Comment:
There are quite a few anonymous functions defined inline in this function component (e.g. L126, L137, L141). Do these need to be memoized with `useCallback`?

Reply:
Done in https://github.com/primer/components/pull/1135/commits/90546176e9e1285fb8b3fc3f69bb19a1d9d69d00